### PR TITLE
DEPR: Passing a dictionary to SeriesGroupBy.agg

### LINKED ISSUE #50684: DEPR: SeriesGroupBy.agg with dict argument
Edit: Instead of implementing the `as_index=True` case mentioned below, the `as_index=False` case should be deprecated, and the docs to SeriesGroupBy.agg should be updated. See the discussion below for details.

According to https://github.com/pandas-dev/pandas/pull/15931#issue-220092580, this was deprecated in 0.20.0 and raises on main now:

```
df = pd.DataFrame({"a": [1, 1, 2], "b": [3, 4, 5]})
gb = df.groupby("a", as_index=True)["b"]
result = gb.agg({"c": "sum"})
print(result)
# pandas.errors.SpecificationError: nested renamer is not supported
```

However, [the docs](https://pandas.pydata.org/pandas-docs/dev/reference/api/pandas.core.groupby.SeriesGroupBy.agg.html) say `SeriesGroupBy.agg` supports dict arguments. Also, when `as_index=False` it works

```
df = pd.DataFrame({"a": [1, 1, 2], "b": [3, 4, 5]})
gb = df.groupby("a", as_index=False)["b"]
result = gb.agg({"c": "sum"})
print(result)
#    a  c
# 0  1  7
# 1  2  5
```

This is because when `as_index=False`, using `__getitem__` with `"b"` still returns a DataFrameGroupBy.

Assuming the implementation in this case isn't difficult, I'm thinking the easiest way forward is to support dictionaries in SeriesGroupBy.agg.

cc @jreback, @jorisvandenbossche

### PR DESCRIPTION
**Pull request by rhshadrach**
- [x] closes #50684 (Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by rhshadrach** [comment]
I don't see a  feasible way to use a single template for both SeriesGroupBy.agg and DataFrameGroupBy.agg, so I split it into two. Is there one, or should these be split?

File: pandas/core/groupby/groupby.py [hunk 1]
```diff
@@ -489,7 +489,94 @@ class providing the base-class of operations.
...
+func : function, str, list, dict or None
+    Function to use for aggregating the data. If a function, must either
+    work when passed a {klass} or when passed to {klass}.apply.
+
+    Accepted combinations are:
+
+    - function
+    - string function name
+    - list of functions and/or function names, e.g. ``[np.sum, 'mean']``
+    - dict of axis labels -> functions, function names or list of such.
```
Comment (mroeschke) [inline pandas/core/groupby/groupby.py:506]: Should we remove noting that dicts can be passed since its deprecated?

Comment (rhshadrach) [reply pandas/core/groupby/groupby.py:506]: For most cases, my preference would be to have the User Guide reflect how we would like users to use pandas (so deprecations are to be removed) and the API docs to be an accurate reflection of the current arguments/behavior (so deprecations should not be removed).

However, in this case `dict` already doesn't work for SeriesGroupBy.agg when `as_index=True` (the default), so this one is a bit of a grey area. I'm good with removing here and will update.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @rhshadrach

**Pull request merged**